sdk/appconfiguration/azure-data-appconfiguration/swagger/src/main/java/AppConfigCustomization.java

```diff
@@ -0,0 +1,146 @@
+            "public SnapshotSettingFilter(String key) {",
+            "   this.key = key;",
+            "}"))
+            .getJavadoc()
+            .setDescription("Constructor of {@link SnapshotSettingFilter}.")
+            .setParam("key", "The key value to set.");
+    }
+
+    private void customizeKeyValueFields(ClassCustomization classCustomization) {
+        classCustomization.addImports("java.util.Locale;");
+        classCustomization.addImports("com.azure.data.appconfiguration.ConfigurationAsyncClient;");
+
+        classCustomization = classCustomization.removeMethod("values");
+
+        classCustomization.getConstant("KEY")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getKey()} from the service.");
+
+        classCustomization.getConstant("LABEL")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLabel()} from the service.");
+
+        classCustomization.getConstant("VALUE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getValue()} from the service.");
+
+        classCustomization.getConstant("CONTENT_TYPE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getContentType()} from the service.");
+
+        classCustomization.getConstant("LAST_MODIFIED")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLastModified()} from the service.");
+
+        classCustomization.getConstant("TAGS")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getTags()} from the service.");
+
+        classCustomization.getConstant("ETAG")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getETag()} from the service.");
+
+        MethodCustomization fromString = classCustomization.getMethod("fromString");
+        fromString.getJavadoc()
+            .setDescription("Creates or finds a {@link SettingFields} from its string representation.")
+            .setReturn("the corresponding {@link SettingFields}");
+        fromString.removeAnnotation("JsonCreator");
+
+        classCustomization = classCustomization.customizeAst(
+            ast -> ast.getClassByName("KeyValueFields")
+                       .get()
+                       .getDefaultConstructor().get()
+                       .remove());
+
+        // Rename LOCKED to IS_READ_ONLY
+        classCustomization = classCustomization.renameEnumMember("LOCKED", "IS_READ_ONLY");
+        classCustomization.getConstant("IS_READ_ONLY")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#isReadOnly()} from the service.");
+
+        // Change class name to SettingFields
+        ClassCustomization settingFields = classCustomization.rename("SettingFields");
```

Comment:
Renaming this doesn't work with the Autorest rename directive?

```
directive:
    - rename-model:
        from: KeyVauleFields
        to: SettingFields
```

Reply:
I thought we prefer to use the customized class instead of directive. I can change it

Reply:
Updated